checkstyle/suppressions.xml

```diff
@@ -294,6 +294,10 @@
     <!-- PhoneHome -->
     <suppress checks="NPathComplexity" files="com[\\/]hazelcast[\\/]util[\\/]PhoneHome"/>
 
+    <!-- SPI -->
+    <suppress checks="FileLength|MethodCount|ClassDataAbstractionCoupling|ClassFanOutComplexity"
```

Comment:
Minor: probably it's better to move this to `AbstractInvocationFuture` itself, if possible. It's hard to keep suppressions up to date if they are specified in a separate file.